python/ray/workflow/step_executor.py

```diff
@@ -149,18 +151,42 @@ def execute_workflow(workflow: "Workflow") -> "WorkflowExecutionResult":
     baked_inputs = _BakedWorkflowInputs.from_workflow_inputs(
         workflow_data.inputs)
     step_options = workflow_data.step_options
-    persisted_output, volatile_output = _workflow_step_executor.options(
-        **step_options.ray_options).remote(
+    if inplace:
+        # TODO(suquark): For inplace execution, it is impossible
+        # to get the ObjectRef of the output before execution.
+        # Here we use a dummy ObjectRef, because _record_step_status does not
+        # even use it (?!).
+        _record_step_status(workflow.step_id, WorkflowStatus.RUNNING,
+                            [ray.put(None)])
```

Comment:
I feel this looks hacky. I feel there might be a bug in the current code, and it should be stored in step cache.

Reply:
In the current code, `_record_step_status` and later calls never uses the third argument. This is very confusing.

Reply:
Agreed. I think there is a bug about this one. Let me check it.

I'm thinking about potential way to do this, but feel ok for now if we have a feature flag in the run.

One way might be, if we can make sure remote task is running in the same node, we probably can use local lock for it. Or we can use db's feature for this, for example, wait until something is there and load it.